File: Easy/hidden-messages-in-images/hidden-messages-in-images.js
var inputs = readline().split(' ');
const w = parseInt(inputs[0]);
const h = parseInt(inputs[1]);
let extracted = "";
for (let i = 0; i < h; i++) {
    var inputs = readline().split(' ');
    for (let j = 0; j < w; j++) {
        const pixel = parseInt(inputs[j]);
        const bin = pixel.toString(2);
        extracted += bin[bin.length-1];
    }
}
while(extracted.length % 8 != 0) extracted = "0" + extracted

let result = "";

for (let i = 0; i < extracted.length; i+=8){
    result += String.fromCharCode(parseInt(extracted.slice(i, i+8), 2));
}
console.log(result); 

File: Easy/hidden-messages-in-images/hidden-messages-in-images.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

class Solution
{
    static void Main(string[] args)
    {
        string[] inputs;
        inputs = Console.ReadLine().Split(' ');
        int w = int.Parse(inputs[0]);
        int h = int.Parse(inputs[1]);
        string extracted = "";
        for (int i = 0; i < h; i++)
        {
            
            inputs = Console.ReadLine().Split(' ');
            for (int j = 0; j < w; j++)
            {
                int pixel = int.Parse(inputs[j]);
                extracted += Convert.ToString(pixel, 2)[^1];
            }
        }
        while (extracted.Count() % 8 != 0) extracted = "0" + extracted;

        byte[] bytes = Enumerable.Range(0, extracted.Count()/8)
                                       .Select(i => Convert.ToByte(extracted.Substring(8 * i, 8), 2))
                                       .ToArray();

        Console.WriteLine(Encoding.ASCII.GetString(bytes));
    }
}